From 2afe7621747c026bf26fa380f6d20c0a1abfd225 Mon Sep 17 00:00:00 2001
From: Sreedhar Govardhana <srgovard@amd.com>
Date: Mon, 16 Aug 2021 18:55:32 -0500
Subject: [PATCH] ARM:dts:aspeed: Adding P0/P1 DIMM SPD slaves to the Quartz
 device tree

This patch will add P0/P1 DIMMs SPDhub slaves to the Quartz device tree. Note
that these slaves added as I2C devices.

Signed-off-by: Sreedhar Govardhana <srgovard@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts | 130 ++++++++++++++++++++
 1 file changed, 130 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index 191afa81a9f1..f826b37319e2 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -252,21 +252,151 @@ &i2c15 {
 &i3c0 {
 	// P0 DIMM (A-F) SPD Access
 	status = "okay";
+
+	jdec-spd;
+
+	//DIMM A
+	spd5118_0_0: spd@50 {
+		compatible = "idt,spd";
+		reg = <0x50 0x0 0x00000000>;
+	};
+	//DIMM B
+	spd5118_0_1: spd@51 {
+		compatible = "idt,spd";
+		reg = <0x51 0x0 0x00000000>;
+	};
+	//DIMM C
+	spd5118_0_2: spd@52 {
+		compatible = "idt,spd";
+		reg = <0x52 0x0 0x00000000>;
+	};
+	//DIMM D
+	spd5118_0_3: spd@53 {
+		compatible = "idt,spd";
+		reg = <0x53 0x0 0x00000000>;
+	};
+	//DIMM E
+	spd5118_0_4: spd@54 {
+		compatible = "idt,spd";
+		reg = <0x54 0x0 0x00000000>;
+	};
+	//DIMM F
+	spd5118_0_5: spd@55 {
+		compatible = "idt,spd";
+		reg = <0x55 0x0 0x00000000>;
+	};
 };

 &i3c1 {
 	// P0 DIMM (G-L) SPD Access
 	status = "okay";
+
+	jdec_spd;
+	//DIMM G
+	spd5118_1_0: spd@50 {
+		compatible = "idt,spd";
+		reg = <0x50 0x0 0x00000000>;
+	};
+	//DIMM H
+	spd5118_1_1: spd@51 {
+		compatible = "idt,spd";
+		reg = <0x51 0x0 0x00000000>;
+	};
+	//DIMM I
+	spd5118_1_2: spd@52 {
+		compatible = "idt,spd";
+		reg = <0x52 0x0 0x00000000>;
+	};
+	//DIMM J
+	spd5118_1_3: spd@53 {
+		compatible = "idt,spd";
+		reg = <0x53 0x0 0x00000000>;
+	};
+	//DIMM K
+	spd5118_1_4: spd@54 {
+		compatible = "idt,spd";
+		reg = <0x54 0x0 0x00000000>;
+	};
+	//DIMM L
+	spd5118_1_5: spd@55 {
+		compatible = "idt,spd";
+		reg = <0x55 0x0 0x00000000>;
+	};
 };

 &i3c2 {
 	// P1 DIMM (A-F) SPD Access
 	status = "okay";
+
+	jdec-spd;
+
+	//DIMM A
+	spd5118_2_0: spd@50 {
+		compatible = "idt,spd";
+		reg = <0x50 0x0 0x00000000>;
+	};
+	//DIMM B
+	spd5118_2_1: spd@51 {
+		compatible = "idt,spd";
+		reg = <0x51 0x0 0x00000000>;
+	};
+	//DIMM C
+	spd5118_2_2: spd@52 {
+		compatible = "idt,spd";
+		reg = <0x52 0x0 0x00000000>;
+	};
+	//DIMM D
+	spd5118_2_3: spd@53 {
+		compatible = "idt,spd";
+		reg = <0x53 0x0 0x00000000>;
+	};
+	//DIMM E
+	spd5118_2_4: spd@54 {
+		compatible = "idt,spd";
+		reg = <0x54 0x0 0x00000000>;
+	};
+	//DIMM F
+	spd5118_2_5: spd@55 {
+		compatible = "idt,spd";
+		reg = <0x55 0x0 0x00000000>;
+	};
 };

 &i3c3 {
 	// P1 DIMM (G-L) SPD Access
 	status = "okay";
+
+	jdec_spd;
+	//DIMM G
+	spd5118_3_0: spd@50 {
+		compatible = "idt,spd";
+		reg = <0x50 0x0 0x00000000>;
+	};
+	//DIMM H
+	spd5118_3_1: spd@51 {
+		compatible = "idt,spd";
+		reg = <0x51 0x0 0x00000000>;
+	};
+	//DIMM I
+	spd5118_3_2: spd@52 {
+		compatible = "idt,spd";
+		reg = <0x52 0x0 0x00000000>;
+	};
+	//DIMM J
+	spd5118_3_3: spd@53 {
+		compatible = "idt,spd";
+		reg = <0x53 0x0 0x00000000>;
+	};
+	//DIMM K
+	spd5118_3_4: spd@54 {
+		compatible = "idt,spd";
+		reg = <0x54 0x0 0x00000000>;
+	};
+	//DIMM L
+	spd5118_3_5: spd@55 {
+		compatible = "idt,spd";
+		reg = <0x55 0x0 0x00000000>;
+	};
 };

 &i3c4 {
--
2.25.1

